src/Functions/formatDateTime.cpp

```diff
@@ -342,12 +342,51 @@ class FunctionFormatDateTimeImpl : public IFunction
             return writeNumber2(dest, ToMonthImpl::execute(source, timezone));
         }
 
+        static size_t mysqlMonthOfYearText(bool abbreviate, char * dest, Time source, UInt64, UInt32, const DateLUTImpl & timezone)
+        {
+            auto month = ToMonthImpl::execute(source, timezone);
+            std::string_view str_view = abbreviate ? monthsShort[month - 1] : monthsFull[month - 1];
+            memcpy(dest, str_view.data(), str_view.size());
+            return str_view.size();
+        }
+
+        static size_t mysqlMonthOfYearTextShort(char * dest, Time source, UInt64 fractional_second, UInt32 scale, const DateLUTImpl & timezone)
+        {
+            return mysqlMonthOfYearText(true, dest, source, fractional_second, scale, timezone);
+        }
+
+        static size_t mysqlMonthOfYearTextLong(char * dest, Time source, UInt64 fractional_second, UInt32 scale, const DateLUTImpl & timezone)
+        {
+            return mysqlMonthOfYearText(false, dest, source, fractional_second, scale, timezone);
+        }
+
         static size_t mysqlDayOfWeek(char * dest, Time source, UInt64, UInt32, const DateLUTImpl & timezone)
         {
             *dest = '0' + ToDayOfWeekImpl::execute(source, 0, timezone);
             return 1;
         }
 
+        static size_t mysqlDayOfWeekText(bool abbreviate, char * dest, Time source, UInt64, UInt32, const DateLUTImpl & timezone)
```

Comment:
Calling "mysql"-prefixed methods from joda methods (specifically: `jodaDayOfWeekText()`) is a little bit odd. What about renaming the method to `dayOfWeekTextImpl()`?

Oh, and for consistency with the other methods, could you please make the `abbreviate` argument the 3rd argument (dest, source, abbreviate)?


Same feedback applies to `mysqlMonthOfYearText()` (l. 345).